geoalchemy2/elements.py

```diff
@@ -131,11 +132,41 @@ class WKBElement(_SpatialElement):
 
     Note: you can create ``WKBElement`` objects from Shapely geometries
     using the :func:`geoalchemy2.shape.from_shape` function.
+
+
+    WKB struct {
+        byte    byteOrder;
+        uint32  wkbType;
+        uint32  SRID;
+        struct  geometry;
+    }
+
+    byteOrder enum {
+        WKB_XDR = 0,  // Most Significant Byte First
+        WKB_NDR = 1,  // Least Significant Byte First
+    }
+
     """
 
     geom_from = 'ST_GeomFromWKB'
     geom_from_extended_version = 'ST_GeomFromEWKB'
 
+    def __init__(self, data, srid=-1, extended=False):
+        if extended and srid == -1:  # unpack srid from WKB struct
+            header = data[:9]
+            try:  # support hex input
+                header = binascii.unhexlify(data[:18])
+            except (binascii.Error, TypeError):
+                pass
+
+            byte_order, srid = header[0], header[5:]
+            if not PY3:
+                byte_order = int(binascii.hexlify(byte_order), 16)
```

Comment:
I'd like to see a comment about why this extra step is necessary with Python 2

Reply:
```python
# bytes string slicing not automatically produces int output in py2
# py2: b'\x01\x00'[0] -> '\x01'
# py3: b'\x01\x00'[0] -> 1
```

Reply:
maybe this is more elegant (and clear):
```python
if not PY3:
    header = bytearray(header)
byte_order, srid = header[0], header[5:]
```

Reply:
Perfect for the comment. Please add it to the code. And +1 for using `bytearray` for Python 2 if that works as expected.

Reply:
done